packages/async-loader/async.js

```diff
@@ -1,36 +1,80 @@
-import { h, Component } from 'preact';
+// import { options, h, Component } from 'preact';
+import { h, Component, options, render } from 'preact';
 
-export default function(req) {
-	function Async() {
-		Component.call(this);
+const f = 'unmount';
+const oldUnmountOpts = options[f];
+const oldDiffed = options.diffed;
 
-		let b, old;
-		this.componentWillMount = () => {
-			b = this.base = this.nextBase || this.__b; // short circuits 1st render
-			req(m => {
-				this.setState({ child: m.default || m });
-			});
-		};
+options[f] = function(vnode) {
+	/**
+	 *  3. now Pending is being unmount and its dom is being removed
+	 *  so we detach the dom from the vnode and empty its children so that Pending is unmount
+	 *  but the DOM is never actually removed.
+	 */
+	if (vnode.type === Pending) {
+		vnode.__e = null;
+		vnode.__k = [];
+	}
+	oldUnmountOpts && oldUnmountOpts(vnode);
+};
+
+options.diffed = function(vnode) {
+	/**
+	 *  4. The route component is now contructed and its DOM will be appended to the browser's DOM.
+	 *  But right before it, we swap its newly contructed DOM with the DOM already present on the browser.
+	 */
+	if (
+		vnode.__e &&
+		vnode.__e.parentNode === null &&
+		vnode.__.__ &&
+		vnode.__.__.type.name === 'AsyncComponent'
```

Comment:
This might be removed by Terser. Could check for a property, or maybe even use the existing `vnode.__.__.type.preload` as a signal that it's AsyncComponent.

Reply:
seems to be working fine tested with build command